src/app/add-item/add-item.component.ts

```diff
@@ -10,59 +10,93 @@ import { first } from 'rxjs/operators';
   templateUrl: './add-item.component.html',
   styleUrls: ['./add-item.component.css']
 })
-export class AddItemComponent implements OnInit {
+export class AddOrUpdateActionItemComponent implements OnInit {
   dialogForm: FormGroup;
-  title = 'Create Action Item';
-  projects: Project[];
-  isCreatingActionItem = false;
+  createDialog: boolean;
+  dialogTitle: string;
+  dialogActionButton: string;
+  projects?: Project[];
+  isSavingDialogData: boolean = false;
+  id: string;
+  loaderVisible: boolean = true;
 
   constructor(
-    public dialogRef: MatDialogRef<AddItemComponent>,
+    public dialogRef: MatDialogRef<AddOrUpdateActionItemComponent>,
     private formBuilder: FormBuilder,
     private dialogProjects: ProjectsService,
-    private tasksService: TasksService
+    private tasksService: TasksService,
+    @Inject(MAT_DIALOG_DATA) public data: any
   ) {}
+
   ngOnInit() {
+    if (this.data.createDialog === true) {
+      this.dialogTitle = 'Create Action Item';
+      this.dialogActionButton = 'Create';
+    } else {
+      this.dialogTitle = 'Edit Action Item';
+      this.dialogActionButton = 'Edit';
+    }
     this.dialogProjects
       .getProjectsNames()
       .pipe(first())
-      .subscribe(projects => (this.projects = projects));
+      .subscribe(projects => {
+        this.projects = projects;
+        this.loaderVisible = false;
+      });
     this.createForm();
+    this.dialogForm.patchValue({
+      title: this.data.item ? this.data.item.title : '',
+      projectName: this.data.item ? this.data.item.projectName : '',
+      dueDate: this.data.item ? this.data.item.dueDate : '',
+      description: this.data.item ? this.data.item.description : '',
+      id: this.data.item ? this.data.item.id : ''
+    });
+    this.id = this.data.id;
+    return this.dialogForm.value;
   }
   private createForm(): void {
     this.dialogForm = this.formBuilder.group({
-      name: ['', Validators.required],
-      project: ['', Validators.required],
+      title: ['', Validators.required],
+      projectName: ['', Validators.required],
       dueDate: '',
-      description: ''
+      description: '',
+      id: ''
     });
   }
+  saveForm(): void {
+    this.isSavingDialogData = true;
+    this.id ? this.editItem() : this.createActionItem();
```

Comment:
**SUGGESTION:** This if should be based on the enum mentioned in https://github.com/romatowskak/examp-ng-app/pull/4/files#r370506748